project/src/main/scala/org/typelevel/idna4s/build/CodePointRange.scala

```diff
@@ -61,8 +126,61 @@ private[build] object CodePointRange {
 
     implicit def orderingInstance: Ordering[Single] =
       hashAndOrderForSingle.toOrdering
+
+    implicit def showInstance: Show[Single] =
+      Show[CodePointRange].narrow[Single]
   }
 
+  /**
+   * Given some `Foldable` of [[CodePointRange]] values mapped to other values of A, create a
+   * `SortedMap` where overlapping values defined later in the `Foldable` replace the overlapped
+   * sections defined earlier.
+   *
+   * For example,
+   * {{{
+   * def a = CodePointRange.unsafeFromInts(0, 10)
+   * def b = CodePointRange.unsafeFromInts(5, 10)
+   * def c = List((a -> "haskell"), (b -> "curry"))
+   * def d = CodePointRange.resolveMissingMapping(c)
+   *
+   * d.foreach(println)
+   *
+   * // yields
+   * // (CodePointRange(CodePoint(value = 0), CodePoint(value = 4)),haskell)
+   * // (CodePointRange(CodePoint(value = 5), CodePoint(value = 10)),curry)
+   * //
+   * // Where the mappings for code points 5 to 10 to "haskell" have been replaced by mappings to "curry".
+   * }}}
+   *
+   * This method is used for handling the Unicode 15.0 behavior of the @missing mapping values.
+   *
+   * @see
+   *   [[https://www.unicode.org/reports/tr44/#Missing_Conventions]]
+   */
+  final def resolveMissingMapping[F[_]: Foldable, A](
+      fa: F[(CodePointRange, A)]): SortedMap[CodePointRange, A] =
+    fa.foldLeft(SortedMap.empty[CodePointRange, A]) {
+      case (acc, (range, a)) =>
+        // Check each current mapping to see if it overlaps with the new,
+        // higher priority, mapping.  This makes this O(n^2). A much more
+        // efficient solution is possible if we had an interval tree data
+        // type. Discussions are in progress about adding one to
+        // cats-collections, but in the mean time we will use this.
+        val value = acc.foldLeft(SortedMap.empty[CodePointRange, A]) {
+          case (acc, (k, v)) if range.overlapsWith(k) =>
+            val (l, r): (Option[CodePointRange], Option[CodePointRange]) = k.difference(range)
+
+            l.fold(acc)(value => acc + (value -> v)) match {
+              case acc =>
+                r.fold(acc)(value => acc + (value -> v))
+            }
```

Comment:
This was a bit of a weird use of match, i think just using a normal variable would be clearer.
```suggestion
            val leftR = l.fold(acc)(value => acc + (value -> v)) 
            r.fold(leftR)(value => leftR + (value -> v))            
```

Reply:
I believe I've addressed the syntax complexity with my most recent commit. It changes the return type of `difference` to be a `Option[Ior[CodePointRange, CodePointRange]]`, which in turn makes this function more simple to write.

Let me know what you think.